subscription.go

```diff
@@ -31,3 +36,24 @@ func (sub *Subscription) Next(ctx context.Context) (*Message, error) {
 func (sub *Subscription) Cancel() {
 	sub.cancelCh <- sub
 }
+
+func (sub *Subscription) validateMsg(ctx context.Context, msg *Message) bool {
+	result := make(chan bool, 1)
+	vctx, cancel := context.WithTimeout(ctx, sub.validateTimeout)
+	defer cancel()
+
+	go func() {
```

Comment:
annoying that we have to use an extra goroutine for this, but probably okay.

Reply:
yeah, that's how it _go_es :)

Reply:
Can we not just trust that validate will respect its context?